### PR TITLE
more doxygen

### LINKED ISSUE #43: Documentation references missing function spgget()
See here: https://noaa-emc.github.io/NCEPLIBS-sp/

There is a reference to function spgget().

But this function does not seem to exist!

### PR DESCRIPTION
**Pull request by edwardhartnett**
Next round of doxygen edits.

Also now upload the build documentation in developer CI, so users can fetch it.

Part of #31
Fixes #43

### CODE REVIEW

**Comment by edwardhartnett** [comment]
Here's the next batch of doxygen improvements for sp. More PRs will be coming...

**Comment by edwardhartnett** [comment]
@AlexanderRichert-NOAA please review so I can merge...

File: src/spfft.f [hunk 1]
```diff
@@ -1,44 +1,42 @@
...
+C> @param IMAX NUMBER OF VALUES IN THE CYCLIC PHYSICAL SPACE
+C> (SEE LIMITATIONS ON IMAX IN REMARKS BELOW.)
+C> @param INCW FIRST DIMENSION OF THE COMPLEX AMPLITUDE ARRAY
+C> (INCW >= IMAX/2+1)
+C> @param INCG FIRST DIMENSION OF THE REAL VALUE ARRAY
+C> (INCG >= IMAX)
+C> @param KMAX NUMBER OF TRANSFORMS TO PERFORM
+C> @param[out] W COMPLEX AMPLITUDES IF IDIR>0
+C> @param[out] G REAL VALUES IF IDIR<0
+C> @param IDIR DIRECTION FLAG
```
Comment (AlexanderRichert-NOAA) [inline src/spfft.f]: capitalization?

File: src/spfft.f [hunk 2]
```diff
@@ -1,44 +1,42 @@
...
+C> - IDIR=0 TO INITIALIZE INTERNAL TRIGONOMETRIC DATA
+C> - IDIR>0 TO TRANSFORM FROM FOURIER TO PHYSICAL SPACE
+C> - IDIR<0 TO TRANSFORM FROM PHYSICAL TO FOURIER SPACE
```
Comment (AlexanderRichert-NOAA) [inline src/spfft.f]: " "

File: src/spgradq.f [hunk 3]
```diff
@@ -21,14 +20,16 @@
...
+C> @param ENN1 
+C> @param ELONN1 
+C> @param EON EPSILON/N*A
+C> @param EONTOP EPSILON/N*A OVER TOP
+C> @param Q SCALAR FIELD
+C> @param QDX ZONAL GRADIENT (TIMES COSLAT)
+C> @param QDY MERID GRADIENT (TIMES COSLAT)
+C> @param QDYTOP MERID GRADIENT (TIMES COSLAT) OVER TOP
+C>      
+C> @author IREDELL @date 92-10-31
```
Comment (AlexanderRichert-NOAA) [inline src/spgradq.f]: capitalization

File: src/sptranf.f [hunk 4]
```diff
@@ -68,6 +45,32 @@
...
+C> @param JS SKIP NUMBER BETWEEN S.H. LATITUDES FROM SOUTH
+C> @param KW SKIP NUMBER BETWEEN WAVE FIELDS
+C> @param KG SKIP NUMBER BETWEEN GRID FIELDS
+C> @param JB LATITUDE INDEX (FROM POLE) TO BEGIN TRANSFORM
+C> @param JE LATITUDE INDEX (FROM POLE) TO END TRANSFORM
+C> @param JC NUMBER OF CPUS OVER WHICH TO MULTIPROCESS
+C> @param[out] WAVE WAVE FIELDS IF IDIR>0
+C> @param[out] GRIDN N.H. GRID FIELDS (STARTING AT JB) IF IDIR<0
+C> @param[out] GRIDS S.H. GRID FIELDS (STARTING AT JB) IF IDIR<0
+C> @param IDIR TRANSFORM FLAG
```
Comment (AlexanderRichert-NOAA) [inline src/sptranf.f]: capitalization

File: src/sptranfv.f [hunk 5]
```diff
@@ -75,6 +45,38 @@
...
+C> @param[out] WAVED WAVE DIVERGENCE FIELDS IF IDIR>0
+C> [WAVED=(D(GRIDU)/DLAM+D(CLAT*GRIDV)/DPHI)/(CLAT*RERTH)]
+C> @param[out] WAVEZ WAVE VORTICITY FIELDS IF IDIR>0
+C> [WAVEZ=(D(GRIDV)/DLAM-D(CLAT*GRIDU)/DPHI)/(CLAT*RERTH)]      
+C> @param[out] GRIDUN N.H. GRID U-WINDS (STARTING AT JB) IF IDIR<0
+C> @param[out] GRIDUS S.H. GRID U-WINDS (STARTING AT JB) IF IDIR<0
+C> @param[out] GRIDVN N.H. GRID V-WINDS (STARTING AT JB) IF IDIR<0
+C> @param[out] GRIDVS S.H. GRID V-WINDS (STARTING AT JB) IF IDIR<0
+C> @param IDIR TRANSFORM FLAG
+C> (IDIR>0 FOR WAVE TO GRID, IDIR<0 FOR GRID TO WAVE)
```
Comment (AlexanderRichert-NOAA) [inline src/sptranfv.f]: capitalization

File: src/sptranfv.f [hunk 6]
```diff
@@ -1,69 +1,39 @@
...
+C>
+C> ### Program History Log
+C> Date | Programmer | Comments
+C> -----|------------|---------
+C> 96-02-29 | IREDELL | Initial.
+C> 1998-12-15 | IREDELL | Generic fft used, openmp directives inserted
+C> 2013-01-16 | IREDELL & MIRVIS | Fixing afft negative sharing effect during omp loops
 C>
-C> Perform a vector spherical transform
 C> @author IREDELL @date 96-02-29
```
Comment (AlexanderRichert-NOAA) [inline src/sptranfv.f]: Author/date should go right after brief, right?

File: src/spfft.f (hunk 1 above)
Comment (edwardhartnett) [reply src/spfft.f]: Fixed.

File: src/spfft.f (hunk 2 above)
Comment (edwardhartnett) [reply src/spfft.f]: Fixed.

File: src/spgradq.f (hunk 3 above)
Comment (edwardhartnett) [reply src/spgradq.f]: Fixed.

File: src/sptranf.f (hunk 4 above)
Comment (edwardhartnett) [reply src/sptranf.f]: Fixed

File: src/sptranfv.f (hunk 5 above)
Comment (edwardhartnett) [reply src/sptranfv.f]: Fixed.

File: src/sptranfv.f (hunk 6 above)
Comment (edwardhartnett) [reply src/sptranfv.f]: No I do:

breif

File History

extra doc (if any)

author/date


Author/date is always last.

Comment (AlexanderRichert-NOAA) [reply src/sptranfv.f]: Got it. I thought we were doing separate author+date entries for file and for each subroutine.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Pull request merged**